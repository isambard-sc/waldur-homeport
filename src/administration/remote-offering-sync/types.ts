export interface RemoteSyncCategoryRule {
  /** UUID of remote category */
  remote_category: string;
  remote_category_name?: string;
  /** URL of local category */
  local_category: string;
  local_category_name?: string;
  local_category_uuid?: string;
}

export interface RemoteSyncForm {
  api_url: string;
  token: string;
  remote_organization_uuid: string;
  remote_organization_name: string;
  /** URL of local service provider */
  local_service_provider: string;
  remotelocalcategory_set: RemoteSyncCategoryRule[];
  is_active: boolean;
}

export interface RemoteSync extends RemoteSyncForm {
  uuid: string;
  url: string;
  last_execution: string;
  last_output: string;
  get_state_display: 'OK' | 'Active' | 'Scheduled';
  error_message: string;
  created: string;
  modified: string;
  local_service_provider_name: string;
}

export interface RemoteSyncActionProps {
  row: RemoteSync;
  refetch;
}
