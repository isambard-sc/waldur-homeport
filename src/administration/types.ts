export interface OIDCConfig {
  auth_url: string;
  client_id: string;
  discovery_url: string;
  enable_pkce: boolean;
  enable_post_logout_redirect: boolean;
  is_active: boolean;
  label: string;
  logout_url: string;
  management_url: string;
  protected_fields: string[];
  provider: string;
  token_url: string;
  userinfo_url: string;
  verify_ssl: boolean;
}
