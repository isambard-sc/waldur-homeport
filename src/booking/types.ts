type DateInput = Date | string | number | number[];

export interface EventInput {
  start?: DateInput;
  end?: DateInput;
  date?: DateInput;
  allDay?: boolean;
  id?: string | number;
  groupId?: string | number;
  title?: string;
  url?: string;
  editable?: boolean;
  rendering?: string;
  extendedProps?: any;
}

export interface BookingState {
  schedules: BookingProps[];
  bookings: BookingProps[];
  config: ConfigProps;
}

export interface ConfigProps {
  weekends: boolean;
  businessHours: {
    startTime: string;
    endTime: string;
    daysOfWeek: number[];
  };
  slotDuration: string;
  minTime?: string;
  maxTime?: string;
  hiddenDays?: number[];
}

export interface BookingProps extends EventInput {
  extendedProps: {
    type: string;
    bookingID?: string;
    config?: ConfigProps;
  };
}

export interface BookingResource {
  uuid: string;
  name: string;
  state: string;
  attributes: {
    schedules: EventInput[];
  };
  customer_name: string;
  customer_uuid: string;
  project_name: string;
  project_description: string;
  provider_uuid: string;
  description: string;
  created: string;
  approved_by_full_name: string;
  created_by_full_name: string;
  created_by_username: string;
}

export interface BookedItem {
  start: string;
  end: string;
  created_by_full_name: string;
}
