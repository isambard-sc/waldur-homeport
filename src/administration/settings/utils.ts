import { titleCase } from '@waldur/core/utils';

export const getKeyTitle = (key) =>
  titleCase(key.toLowerCase().replaceAll('_', ' '));
