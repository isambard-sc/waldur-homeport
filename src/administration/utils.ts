import { translate } from '@waldur/i18n';

export const AnnouncementTypeOptions = [
  {
    label: translate('Information'),
    value: 'information',
  },
  {
    label: translate('Warning'),
    value: 'warning',
  },
  {
    label: translate('Danger'),
    value: 'danger',
  },
];
