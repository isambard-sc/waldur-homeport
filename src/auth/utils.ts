import Qs from 'qs';

import { EDUTEAMS_IDP } from './providers/constants';

const getStateId = () =>
  encodeURIComponent(Math.random().toString(36).substring(2));

export const getRedirectUri = (provider) =>
  `${document.baseURI}oauth_login_completed/${provider}/`;

export const getOauthURL = (provider) =>
  `${provider.auth_url}?${Qs.stringify({
    response_type: 'code',
    client_id: provider.client_id,
    redirect_uri: getRedirectUri(provider.provider),
    scope:
      provider.provider === EDUTEAMS_IDP
        ? 'openid profile email eduperson_assurance ssh_public_key'
        : 'openid',
    state: getStateId(),
  })}`;
