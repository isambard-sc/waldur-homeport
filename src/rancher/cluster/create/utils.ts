import { useQuery } from '@tanstack/react-query';
import { formValueSelector } from 'redux-form';

import { ENV } from '@waldur/configs/default';
import { translate } from '@waldur/i18n';
import { ORDER_FORM_ID } from '@waldur/marketplace/details/constants';
import { Offering } from '@waldur/marketplace/types';
import {
  loadFlavors,
  loadSubnets,
  loadVolumeTypes,
  loadSecurityGroups,
} from '@waldur/openstack/api';
import { Flavor, Subnet } from '@waldur/openstack/openstack-instance/types';
import {
  formatVolumeTypeChoices,
  getDefaultVolumeType,
} from '@waldur/openstack/openstack-instance/utils';
import { listClusterTemplates } from '@waldur/rancher/api';
import { Cluster, NodeField } from '@waldur/rancher/types';
import { formatFlavor } from '@waldur/resource/utils';
import { RootState } from '@waldur/store/reducers';

const CLUSTER_NAME_PATTERN = new RegExp('^[a-z0-9]([-a-z0-9])+[a-z0-9]$');

export const rancherClusterName = (value: string) =>
  !value.match(CLUSTER_NAME_PATTERN)
    ? translate('Name must consist of lower case alphanumeric characters.')
    : undefined;

const formatSubnetOption = (subnet: Subnet) => ({
  label: `${subnet.network_name} / ${subnet.name} (${subnet.cidr})`,
  value: subnet.url,
});

const formatFlavorOption = (flavor: Flavor) => ({
  ...flavor,
  label: `${flavor.name} (${formatFlavor(flavor)})`,
  value: flavor.url,
});

export const filterFlavors = (tenant_uuid: string, offering: Offering) => {
  return loadFlavors({
    tenant_uuid,
    name_iregex: offering.plugin_options?.flavors_regex,
  }).then((data) => data.map(formatFlavorOption));
};

export const formatSubnets = (tenant_uuid: string) =>
  loadSubnets({ tenant_uuid }).then((data) => data.map(formatSubnetOption));

export const getRancherMountPointChoices = () => {
  const mountPoints = ENV.plugins.WALDUR_RANCHER.MOUNT_POINT_CHOICES;
  return mountPoints.map((choice) => ({
    label: choice,
    value: choice,
  }));
};

export const loadData = async (cluster: Cluster, offering: Offering) => {
  const flavors = await filterFlavors(cluster.tenant_uuid, offering);
  const subnets = await formatSubnets(cluster.tenant_uuid);
  const volumeTypes = await loadVolumeTypes({
    tenant_uuid: cluster.tenant_uuid,
  });
  const templates = await listClusterTemplates();
  const volumeTypeChoices = formatVolumeTypeChoices(volumeTypes);
  const defaultVolumeType = getDefaultVolumeType(volumeTypeChoices);
  const securityGroups = await loadSecurityGroups({
    tenant_uuid: cluster.tenant_uuid,
  });
  return {
    subnets,
    flavors,
    volumeTypes: volumeTypeChoices,
    defaultVolumeType: defaultVolumeType && defaultVolumeType.value,
    mountPoints: getRancherMountPointChoices(),
    templates,
    securityGroups,
  };
};

export const useVolumeDataLoader = (tenantSettings) => {
  return useQuery(
    ['volumeTypes', tenantSettings],
    async () => {
      const volumeTypes = tenantSettings
        ? await loadVolumeTypes({ tenant: tenantSettings.scope })
        : [];
      const volumeTypeChoices = formatVolumeTypeChoices(volumeTypes);
      const defaultVolumeType = getDefaultVolumeType(volumeTypeChoices);
      return {
        volumeTypeChoices,
        defaultVolumeType,
      };
    },
    { staleTime: 3 * 60 * 1000 },
  );
};

export const getDataVolumes = (nodeIndex, allValues) => {
  if (nodeIndex !== undefined) {
    const nodes = allValues.attributes.nodes;
    if (nodeIndex >= nodes.length) {
      return [];
    }
    return nodes[nodeIndex].data_volumes || [];
  } else {
    return allValues.data_volumes || [];
  }
};

export const formTenantSelector = (state: RootState) =>
  formValueSelector(ORDER_FORM_ID)(state, 'attributes.tenant_settings');

export const formNodesSelector = (state: RootState): NodeField[] =>
  formValueSelector(ORDER_FORM_ID)(state, 'attributes.nodes');
