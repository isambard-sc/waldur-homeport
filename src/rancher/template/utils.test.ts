import { describe, it, expect } from 'vitest';

import { Question } from '../types';

import DataDog from './DataDog.json';
import JFrog from './JFrog.json';
import MySQL from './MySQL.json';
import { groupQuestions, parseQuestions, parseVisibleQuestions } from './utils';

describe('Rancher application provision utils', () => {
  describe('groupQuestions', () => {
    it('groups questions', () => {
      expect(
        Object.keys(groupQuestions(DataDog as Question[])),
      ).toMatchSnapshot();
    });
  });

  describe('subquestions', () => {
    it('shows subquestions if condition is met', () => {
      const parsed = parseQuestions(MySQL as Question[]);
      const visible = parseVisibleQuestions(parsed, { defaultImage: false });
      expect(
        visible.find((question) => question.variable === 'image'),
      ).toBeTruthy();
    });

    it('hides subquestions if condition is not met', () => {
      const parsed = parseQuestions(MySQL as Question[]);
      const visible = parseVisibleQuestions(parsed, { defaultImage: true });
      expect(
        visible.find((question) => question.variable === 'image'),
      ).toBeFalsy();
    });

    it('parses subquestion types', () => {
      const parsed = parseQuestions(JFrog as Question[]);
      expect(parsed).toMatchSnapshot();
    });
  });
});
