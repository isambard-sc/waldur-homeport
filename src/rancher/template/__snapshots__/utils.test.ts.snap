// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Rancher application provision utils > groupQuestions > groups questions 1`] = `
[
  "Container Images",
  "Cluster Agent",
  "Agent Configuration",
  "Datadog Tagging",
  "Daemonset Configuration",
  "Proxy Configuration",
  "Optional Collection Agents",
  "Kube-State-Metrics",
]
`;

exports[`Rancher application provision utils > subquestions > parses subquestion types 1`] = `
[
  {
    "default": "true",
    "description": "Use default Docker image",
    "group": "Container Images",
    "label": "Use Default Image",
    "showIf": undefined,
    "showSubquestionIf": false,
    "subquestions": [
      {
        "default": "alpine:3.10",
        "description": "Init image name",
        "group": "Container Images",
        "label": "Init image name",
        "showIf": undefined,
        "type": "string",
        "variable": "initContainerImage",
      },
      {
        "default": "docker.bintray.io/jfrog/artifactory-pro",
        "description": "Artifactory image name",
        "group": "Container Images",
        "label": "Artifactory Image Name",
        "showIf": undefined,
        "type": "string",
        "variable": "artifactory.image.repository",
      },
      {
        "default": "6.13.1",
        "description": "Artifactory image tag",
        "group": "Container Images",
        "label": "Artifactory Image Tag",
        "showIf": undefined,
        "type": "string",
        "variable": "artifactory.image.version",
      },
      {
        "default": "docker.bintray.io/jfrog/nginx-artifactory-pro",
        "description": "Nginx image name",
        "group": "Container Images",
        "label": "Nginx Image Name",
        "showIf": undefined,
        "type": "string",
        "variable": "nginx.image.repository",
      },
      {
        "default": "6.13.1",
        "description": "Nginx image tag",
        "group": "Container Images",
        "label": "Nginx Image Tag",
        "showIf": undefined,
        "type": "string",
        "variable": "nginx.image.version",
      },
      {
        "description": "Image Pull Secret",
        "group": "Container Images",
        "label": "Image Pull Secret",
        "showIf": undefined,
        "type": "string",
        "variable": "imagePullSecrets",
      },
    ],
    "type": "boolean",
    "variable": "defaultImage",
  },
  {
    "default": "true",
    "description": "Expose app using Layer 7 Load Balancer - ingress",
    "group": "Services and Load Balancing",
    "label": "Expose app using Layer 7 Load Balancer",
    "required": true,
    "showIf": undefined,
    "showSubquestionIf": true,
    "subquestions": [
      {
        "default": "xip.io",
        "description": "Hostname to your artifactory installation",
        "group": "Services and Load Balancing",
        "label": "Hostname",
        "required": true,
        "showIf": undefined,
        "type": "hostname",
        "variable": "ingress.hosts[0]",
      },
    ],
    "type": "boolean",
    "variable": "ingress.enabled",
  },
  {
    "default": "false",
    "description": "Enable nginx server",
    "group": "Services and Load Balancing",
    "label": "Enable Nginx Server",
    "required": true,
    "showIf": {
      "ingress.enabled": false,
    },
    "showSubquestionIf": undefined,
    "subquestions": undefined,
    "type": "boolean",
    "variable": "nginx.enabled",
  },
  {
    "default": "LoadBalancer",
    "description": "Nginx service type",
    "group": "Services and Load Balancing",
    "label": "Nginx Service Type",
    "options": [
      "ClusterIP",
      "NodePort",
      "LoadBalancer",
    ],
    "required": true,
    "showIf": {
      "ingress.enabled": false,
      "nginx.enabled": true,
    },
    "showSubquestionIf": undefined,
    "subquestions": undefined,
    "type": "enum",
    "variable": "nginx.service.type",
  },
  {
    "description": "Provide Static IP to configure with Nginx",
    "group": "Services and Load Balancing",
    "label": "Config Nginx LoadBalancer IP",
    "showIf": {
      "ingress.enabled": false,
      "nginx.enabled": true,
      "nginx.service.type": "LoadBalancer",
    },
    "showSubquestionIf": undefined,
    "subquestions": undefined,
    "type": "string",
    "variable": "nginx.service.loadBalancerIP",
  },
  {
    "description": "Provide SSL Secret name to configure with Nginx",
    "group": "Services and Load Balancing",
    "label": "Config Nginx SSL Secret",
    "showIf": {
      "ingress.enabled": false,
      "nginx.enabled": true,
    },
    "showSubquestionIf": undefined,
    "subquestions": undefined,
    "type": "string",
    "variable": "nginx.tlsSecretName",
  },
  {
    "description": "Provide configMap name to configure Nginx with custom \`artifactory.conf\`",
    "group": "Services and Load Balancing",
    "label": "ConfigMap for Nginx Artifactory Config",
    "showIf": {
      "ingress.enabled": false,
      "nginx.enabled": true,
    },
    "showSubquestionIf": undefined,
    "subquestions": undefined,
    "type": "string",
    "variable": "nginx.customArtifactoryConfigMap",
  },
  {
    "default": "false",
    "description": "Skip Nginx Auto Configuration Update",
    "group": "Services and Load Balancing",
    "label": "Skip Nginx Auto Configuration Update",
    "showIf": {
      "ingress.enabled": false,
      "nginx.enabled": true,
    },
    "showSubquestionIf": undefined,
    "subquestions": undefined,
    "type": "string",
    "variable": "nginx.env.skipAutoConfigUpdate",
  },
  {
    "default": "false",
    "description": "enable persistence storage for nginx server",
    "group": "Services and Load Balancing",
    "label": "Enable Persistent Storage for Nginx Service",
    "showIf": {
      "ingress.enabled": false,
      "nginx.enabled": true,
    },
    "showSubquestionIf": undefined,
    "subquestions": undefined,
    "type": "boolean",
    "variable": "nginx.persistence.enabled",
  },
  {
    "default": "5Gi",
    "description": "Nginx persistent volume size",
    "group": "Services and Load Balancing",
    "label": "Nginx Persistent Volume Size",
    "showIf": {
      "ingress.enabled": false,
      "nginx.enabled": true,
      "nginx.persistence.enabled": true,
    },
    "showSubquestionIf": undefined,
    "subquestions": undefined,
    "type": "string",
    "variable": "nginx.persistence.size",
  },
  {
    "description": "If undefined or null, uses the default StorageClass. Default to null",
    "group": "Services and Load Balancing",
    "label": "Default StorageClass for Nginx",
    "showIf": {
      "ingress.enabled": false,
      "nginx.enabled": true,
      "nginx.persistence.enabled": true,
    },
    "showSubquestionIf": undefined,
    "subquestions": undefined,
    "type": "storageclass",
    "variable": "nginx.persistence.storageClass",
  },
  {
    "default": "200m",
    "description": "Nginx initial cpu request",
    "group": "Services and Load Balancing",
    "label": "Nginx Initial CPU Request",
    "showIf": {
      "ingress.enabled": false,
      "nginx.enabled": true,
    },
    "showSubquestionIf": undefined,
    "subquestions": undefined,
    "type": "string",
    "variable": "nginx.resources.requests.cpu",
  },
  {
    "default": "500Mi",
    "description": "Nginx initial memory request",
    "group": "Services and Load Balancing",
    "label": "Nginx Initial Memory Request",
    "showIf": {
      "ingress.enabled": false,
      "nginx.enabled": true,
    },
    "showSubquestionIf": undefined,
    "subquestions": undefined,
    "type": "string",
    "variable": "nginx.resources.requests.memory",
  },
  {
    "default": "1",
    "description": "Nginx cpu limit",
    "group": "Services and Load Balancing",
    "label": "Nginx CPU Limit",
    "showIf": {
      "ingress.enabled": false,
      "nginx.enabled": true,
    },
    "showSubquestionIf": undefined,
    "subquestions": undefined,
    "type": "string",
    "variable": "nginx.resources.limits.cpu",
  },
  {
    "default": "1Gi",
    "description": "Nginx memory limit",
    "group": "Services and Load Balancing",
    "label": "Nginx Memory Limit",
    "showIf": {
      "ingress.enabled": false,
      "nginx.enabled": true,
    },
    "showSubquestionIf": undefined,
    "subquestions": undefined,
    "type": "string",
    "variable": "nginx.resources.limits.memory",
  },
  {
    "default": "true",
    "description": "enable persistence storage for Artifactory",
    "group": "Artifactory Storage",
    "label": "Enable Persistent Storage for Artifactory",
    "required": true,
    "showIf": undefined,
    "showSubquestionIf": true,
    "subquestions": [
      {
        "default": "50Gi",
        "description": "Artifactory persistent volume size",
        "group": "Artifactory Storage",
        "label": "Artifactory Persistent Volume Size",
        "required": true,
        "showIf": undefined,
        "type": "string",
        "variable": "artifactory.persistence.size",
      },
      {
        "default": "file-system",
        "description": "Artifactory persistent volume size",
        "group": "Artifactory Storage",
        "label": "Artifactory Persistent Storage Type",
        "options": [
          "file-system",
          "nfs",
          "google-storage",
          "aws-s3",
        ],
        "required": true,
        "showIf": undefined,
        "type": "enum",
        "variable": "artifactory.persistence.type",
      },
    ],
    "type": "boolean",
    "variable": "artifactory.persistence.enabled",
  },
  {
    "description": "NFS server IP",
    "group": "Artifactory Storage",
    "label": "NFS Server IP",
    "showIf": {
      "artifactory.persistence.enabled": true,
      "artifactory.persistence.type": "nfs",
    },
    "showSubquestionIf": undefined,
    "subquestions": undefined,
    "type": "string",
    "variable": "artifactory.persistence.nfs.ip",
  },
  {
    "default": "/data",
    "description": "NFS data directory",
    "group": "Artifactory Storage",
    "label": "NFS Data Directory",
    "showIf": {
      "artifactory.persistence.enabled": true,
      "artifactory.persistence.type": "nfs",
    },
    "showSubquestionIf": undefined,
    "subquestions": undefined,
    "type": "string",
    "variable": "artifactory.persistence.nfs.haDataMount",
  },
  {
    "default": "/backup",
    "description": "NFS backup directory",
    "group": "Artifactory Storage",
    "label": "NFS Backup Directory",
    "showIf": {
      "artifactory.persistence.enabled": true,
      "artifactory.persistence.type": "nfs",
    },
    "showSubquestionIf": undefined,
    "subquestions": undefined,
    "type": "string",
    "variable": "artifactory.persistence.nfs.haBackupMount",
  },
  {
    "default": "/var/opt/jfrog/artifactory-ha",
    "description": "HA data directory",
    "group": "Artifactory Storage",
    "label": "HA Data Directory",
    "showIf": {
      "artifactory.persistence.enabled": true,
      "artifactory.persistence.type": "nfs",
    },
    "showSubquestionIf": undefined,
    "subquestions": undefined,
    "type": "string",
    "variable": "artifactory.persistence.nfs.dataDir",
  },
  {
    "default": "/var/opt/jfrog/artifactory-backup",
    "description": "HA backup directory	",
    "group": "Artifactory Storage",
    "label": "HA Backup Directory",
    "showIf": {
      "artifactory.persistence.enabled": true,
      "artifactory.persistence.type": "nfs",
    },
    "showSubquestionIf": undefined,
    "subquestions": undefined,
    "type": "string",
    "variable": "artifactory.persistence.nfs.backupDir",
  },
  {
    "default": "200Gi",
    "description": "NFS PVC size	",
    "group": "Artifactory Storage",
    "label": "NFS PVC Size",
    "showIf": {
      "artifactory.persistence.enabled": true,
      "artifactory.persistence.type": "nfs",
    },
    "showSubquestionIf": undefined,
    "subquestions": undefined,
    "type": "string",
    "variable": "artifactory.persistence.nfs.capacity",
  },
  {
    "default": "artifactory-ha-gcp",
    "description": "Google storage bucket name",
    "group": "Artifactory Storage",
    "label": "Google Storage Bucket Name",
    "showIf": {
      "artifactory.persistence.enabled": true,
      "artifactory.persistence.type": "google-storage",
    },
    "showSubquestionIf": undefined,
    "subquestions": undefined,
    "type": "string",
    "variable": "artifactory.persistence.googleStorage.bucketName",
  },
  {
    "description": "Google Storage service account id",
    "group": "Artifactory Storage",
    "label": "Google Storage Service Account ID",
    "showIf": {
      "artifactory.persistence.enabled": true,
      "artifactory.persistence.type": "google-storage",
    },
    "showSubquestionIf": undefined,
    "subquestions": undefined,
    "type": "string",
    "variable": "artifactory.persistence.googleStorage.identity",
  },
  {
    "description": "Google Storage service account key",
    "group": "Artifactory Storage",
    "label": "Google Storage Service Account Key",
    "showIf": {
      "artifactory.persistence.enabled": true,
      "artifactory.persistence.type": "google-storage",
    },
    "showSubquestionIf": undefined,
    "subquestions": undefined,
    "type": "string",
    "variable": "artifactory.persistence.googleStorage.credential",
  },
  {
    "default": "artifactory-ha/filestore",
    "description": "Google Storage path in bucket",
    "group": "Artifactory Storage",
    "label": "Google Storage Path In Bucket",
    "showIf": {
      "artifactory.persistence.enabled": true,
      "artifactory.persistence.type": "google-storage",
    },
    "showSubquestionIf": undefined,
    "subquestions": undefined,
    "type": "string",
    "variable": "artifactory.persistence.googleStorage.path",
  },
  {
    "default": "artifactory-ha-aws",
    "description": "AWS S3 bucket name",
    "group": "Artifactory Storage",
    "label": "AWS S3 Bucket Name",
    "showIf": {
      "artifactory.persistence.enabled": true,
      "artifactory.persistence.type": "aws-s3",
    },
    "showSubquestionIf": undefined,
    "subquestions": undefined,
    "type": "string",
    "variable": "artifactory.persistence.awsS3.bucketName",
  },
  {
    "description": "AWS S3 bucket region",
    "group": "Artifactory Storage",
    "label": "AWS S3 Bucket Region",
    "showIf": {
      "artifactory.persistence.enabled": true,
      "artifactory.persistence.type": "aws-s3",
    },
    "showSubquestionIf": undefined,
    "subquestions": undefined,
    "type": "string",
    "variable": "artifactory.persistence.awsS3.region",
  },
  {
    "description": "AWS S3 AWS_ACCESS_KEY_ID",
    "group": "Artifactory Storage",
    "label": "AWS S3 AWS_ACCESS_KEY_ID",
    "showIf": {
      "artifactory.persistence.enabled": true,
      "artifactory.persistence.type": "aws-s3",
    },
    "showSubquestionIf": undefined,
    "subquestions": undefined,
    "type": "string",
    "variable": "artifactory.persistence.awsS3.identity",
  },
  {
    "description": "AWS S3 AWS_SECRET_ACCESS_KEY",
    "group": "Artifactory Storage",
    "label": "AWS S3 AWS_SECRET_ACCESS_KEY",
    "showIf": {
      "artifactory.persistence.enabled": true,
      "artifactory.persistence.type": "aws-s3",
    },
    "showSubquestionIf": undefined,
    "subquestions": undefined,
    "type": "string",
    "variable": "artifactory.persistence.awsS3.credential",
  },
  {
    "default": "artifactory-ha/filestore",
    "description": "AWS S3 path in bucket",
    "group": "Artifactory Storage",
    "label": "AWS S3 Path In Bucket",
    "showIf": {
      "artifactory.persistence.enabled": true,
      "artifactory.persistence.type": "aws-s3",
    },
    "showSubquestionIf": undefined,
    "subquestions": undefined,
    "type": "string",
    "variable": "artifactory.persistence.awsS3.path",
  },
  {
    "default": "true",
    "description": "Enable PostgreSQL",
    "group": "Database Settings",
    "label": "Enable PostgreSQL",
    "required": true,
    "showIf": undefined,
    "showSubquestionIf": true,
    "subquestions": [
      {
        "default": "false",
        "description": "enable persistence storage for PostgreSQL",
        "group": "Database Settings",
        "label": "Enable Persistent Storage for PostgreSQL",
        "showIf": undefined,
        "type": "boolean",
        "variable": "postgresql.persistence.enabled",
      },
      {
        "default": "20Gi",
        "description": "PostgreSQL persistent volume size",
        "group": "Database Settings",
        "label": "PostgreSQL Persistent Volume Size",
        "showIf": {
          "postgresql.enabled": true,
          "postgresql.persistence.enabled": true,
        },
        "type": "string",
        "variable": "postgresql.persistence.size",
      },
      {
        "description": "If undefined or null, uses the default StorageClass. Default to null",
        "group": "Database Settings",
        "label": "Default StorageClass for PostgreSQL",
        "showIf": {
          "postgresql.enabled": true,
          "postgresql.persistence.enabled": true,
        },
        "type": "storageclass",
        "variable": "postgresql.persistence.storageClass",
      },
      {
        "default": "200m",
        "description": "PostgreSQL initial cpu request",
        "group": "Database Settings",
        "label": "PostgreSQL Initial CPU Request",
        "showIf": {
          "postgresql.enabled": true,
        },
        "type": "string",
        "variable": "postgresql.resources.requests.cpu",
      },
      {
        "default": "500Mi",
        "description": "PostgreSQL initial memory request",
        "group": "Database Settings",
        "label": "PostgreSQL Initial Memory Request",
        "showIf": {
          "postgresql.enabled": true,
        },
        "type": "string",
        "variable": "postgresql.resources.requests.memory",
      },
      {
        "default": "1",
        "description": "PostgreSQL cpu limit",
        "group": "Database Settings",
        "label": "PostgreSQL CPU Limit",
        "showIf": {
          "postgresql.enabled": true,
        },
        "type": "string",
        "variable": "postgresql.resources.limits.cpu",
      },
      {
        "default": "1Gi",
        "description": "PostgreSQL memory limit",
        "group": "Database Settings",
        "label": "PostgreSQL Memory Limit",
        "showIf": {
          "postgresql.enabled": true,
        },
        "type": "string",
        "variable": "postgresql.resources.limits.memory",
      },
    ],
    "type": "boolean",
    "variable": "postgresql.enabled",
  },
  {
    "default": "postgresql",
    "description": "xternal database type (postgresql, mysql, oracle or mssql)",
    "group": "Database Settings",
    "label": "External Database Type",
    "options": [
      "postgresql",
      "mysql",
      "oracle",
      "mssql",
    ],
    "required": true,
    "showIf": {
      "postgresql.enabled": false,
    },
    "showSubquestionIf": undefined,
    "subquestions": undefined,
    "type": "enum",
    "variable": "database.type",
  },
  {
    "description": "External database URL. If you set the url, leave host and port empty",
    "group": "Database Settings",
    "label": "External Database URL",
    "showIf": {
      "postgresql.enabled": false,
    },
    "showSubquestionIf": undefined,
    "subquestions": undefined,
    "type": "string",
    "variable": "database.url",
  },
  {
    "description": "External database hostname",
    "group": "Database Settings",
    "label": "External Database Hostname",
    "showIf": {
      "postgresql.enabled": false,
    },
    "showSubquestionIf": undefined,
    "subquestions": undefined,
    "type": "string",
    "variable": "database.host",
  },
  {
    "description": "External database port",
    "group": "Database Settings",
    "label": "External Database Port",
    "showIf": {
      "postgresql.enabled": false,
    },
    "showSubquestionIf": undefined,
    "subquestions": undefined,
    "type": "string",
    "variable": "database.port",
  },
  {
    "description": "External database username",
    "group": "Database Settings",
    "label": "External Database Username",
    "showIf": {
      "postgresql.enabled": false,
    },
    "showSubquestionIf": undefined,
    "subquestions": undefined,
    "type": "string",
    "variable": "database.user",
  },
  {
    "description": "External database password",
    "group": "Database Settings",
    "label": "External Database Password",
    "showIf": {
      "postgresql.enabled": false,
    },
    "showSubquestionIf": undefined,
    "subquestions": undefined,
    "type": "string",
    "variable": "database.password",
  },
  {
    "default": "false",
    "description": "Show advanced configurations",
    "group": "Advanced Options",
    "label": "Show Advanced Configurations",
    "showIf": undefined,
    "showSubquestionIf": true,
    "subquestions": [
      {
        "default": "FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF",
        "description": "Artifactory master key",
        "group": "Advanced Options",
        "label": "Artifactory master key",
        "showIf": undefined,
        "type": "string",
        "variable": "artifactory.masterKey",
      },
      {
        "default": "500m",
        "description": "Artifactory primary node initial cpu request",
        "group": "Advanced Options",
        "label": "Artifactory Primary Node Initial CPU Request",
        "showIf": undefined,
        "type": "string",
        "variable": "artifactory.primary.resources.requests.cpu",
      },
      {
        "default": "1Gi",
        "description": "Artifactory primary node initial memory request",
        "group": "Advanced Options",
        "label": "Artifactory Primary Node Initial Memory Request",
        "showIf": undefined,
        "type": "string",
        "variable": "artifactory.primary.resources.requests.memory",
      },
      {
        "default": "1g",
        "description": "Artifactory primary node java Xms size",
        "group": "Advanced Options",
        "label": "Artifactory Primary Node Java Xms Size",
        "showIf": undefined,
        "type": "string",
        "variable": "artifactory.primary.javaOpts.xms",
      },
      {
        "default": "2",
        "description": "Artifactory primary node cpu limit",
        "group": "Advanced Options",
        "label": "Artifactory Primary Node CPU Limit",
        "showIf": undefined,
        "type": "string",
        "variable": "artifactory.primary.resources.limits.cpu",
      },
      {
        "default": "4Gi",
        "description": "Artifactory primary node memory limit",
        "group": "Advanced Options",
        "label": "Artifactory Primary Node Memory Limit",
        "showIf": undefined,
        "type": "string",
        "variable": "artifactory.primary.resources.limits.memory",
      },
      {
        "default": "4g",
        "description": "Artifactory primary node java Xmx size",
        "group": "Advanced Options",
        "label": "Artifactory Primary Node Java Xmx Size",
        "showIf": undefined,
        "type": "string",
        "variable": "artifactory.primary.javaOpts.xmx",
      },
      {
        "default": "500m",
        "description": "Artifactory member node initial cpu request",
        "group": "Advanced Options",
        "label": "Artifactory Member Node Initial CPU Request",
        "showIf": undefined,
        "type": "string",
        "variable": "artifactory.node.resources.requests.cpu",
      },
      {
        "default": "2Gi",
        "description": "Artifactory member node initial memory request",
        "group": "Advanced Options",
        "label": "Artifactory Member Node Initial Memory Request",
        "showIf": undefined,
        "type": "string",
        "variable": "artifactory.node.resources.requests.memory",
      },
      {
        "default": "1g",
        "description": "Artifactory member node java Xms size",
        "group": "Advanced Options",
        "label": "Artifactory Member Node Java Xms Size",
        "showIf": undefined,
        "type": "string",
        "variable": "artifactory.node.javaOpts.xms",
      },
      {
        "default": "2",
        "description": "Artifactory member node cpu limit",
        "group": "Advanced Options",
        "label": "Artifactory Member Node CPU Limit",
        "showIf": undefined,
        "type": "string",
        "variable": "artifactory.node.resources.limits.cpu",
      },
      {
        "default": "4Gi",
        "description": "Artifactory member node memory limit",
        "group": "Advanced Options",
        "label": "Artifactory Member Node Memory Limit",
        "showIf": undefined,
        "type": "string",
        "variable": "artifactory.node.resources.limits.memory",
      },
      {
        "default": "4g",
        "description": "Artifactory member node java Xmx size",
        "group": "Advanced Options",
        "label": "Artifactory Member Node Java Xmx Size",
        "showIf": undefined,
        "type": "string",
        "variable": "artifactory.node.javaOpts.xmx",
      },
    ],
    "type": "boolean",
    "variable": "advancedOptions",
  },
]
`;
